### PR TITLE
Migrate Button component to semantic color tokens

### PR DESCRIPTION
**Pull request by robinmetral**
🚧 POC 🚧 

Depends on #1880 

## Purpose

For now, this is intended as a POC.

It migrates the `Button` component (quite color-heavy and used as an atom in many other components) to use semantic color tokens instead of the legacy theme.

## Approach and changes

_Describe how you solved the problem_

## Definition of done

* [ ] Development completed
* [ ] Reviewers assigned
* [ ] Unit and integration tests
* [ ] Meets minimum browser support
* [ ] Meets accessibility requirements

### CODE REVIEW

File: packages/circuit-ui/components/Button/Button.tsx [hunk 1]
```diff
@@ -157,34 +157,93 @@ const Content = styled.span<{ isLoading: boolean }>(
   contentLoadingStyles,
 );
 
-const COLOR_MAP = {
+const PRIMARY_COLOR_MAP = {
```
Comment (connor-baer) [inline packages/circuit-ui/components/Button/Button.tsx:160]: Now that we're using CSS variables, I'd refactor this code to move away from this object and replace it with a `switch` statement and explicit styles. Would help readability I think.

Comment (robinmetral) [reply packages/circuit-ui/components/Button/Button.tsx:160]: Good idea! Just replaced the tokens for now, but you're right that this is a good opportunity to make this a bit clearer. Will address (in January)

Comment (connor-baer) [reply packages/circuit-ui/components/Button/Button.tsx:160]: Was contemplating this again and realized it's actually a matrix between `variant` and `destructive`. Would keep it for now and refactor with the new button design.

**Comment by connor-baer** [comment]
This PR was merged locally into #1944 and has been released as part of Circuit UI v6.3.

**Pull request closed without merging**